backend/studytimeboard/routes.py

```diff
@@ -148,15 +148,32 @@ def api_handle_go_event():
 @app.route("/api/hold", methods=["POST"])
 def api_handle_hold_event():
     data = json.loads(request.data)
-    username = data["username"]
+    username = data[USERNAME]
     if username in dbapi.all_users():
         date = datetime.now(TZ)
         end_time = datetime2time(datetime.now(TZ))
         dbapi.into_hold(username, date, end_time)
         return {"status": "success"}, 200
     else:
         return {"status": "error", "message": FlashMessages.NO_SUCH_USER}, 400
+    
+@app.route("/api/duration", methods=["POST"])
+def api_handle_duration_event():
+    data = json.loads(request.data)
+    username = data[USERNAME]
+    if username in dbapi.all_users():
+        date = datetime.now(TZ)
+        start_time = data[START_TIME]
+        end_time = data[END_TIME]
 
+        if varify_time(start_time) and varify_time(end_time):
+            dbapi.into_duration(username, date, start_time, end_time)
+            print(username, start_time, end_time)
```

Comment:
- [ ] this print seems to be there only for debug and not logging. Should be removed before merging